web/server/entrypoint_test.go

```diff
@@ -156,7 +160,14 @@ func TestShutdown(t *testing.T) {
 		rc := robotpb.NewRobotServiceClient(conn)
 
 		_, err = rc.Shutdown(context.Background(), &robotpb.ShutdownRequest{})
-		test.That(t, err, test.ShouldBeNil)
+
+		gtestutils.WaitForAssertionWithSleep(t, 50*time.Millisecond, 50, func(tb testing.TB) {
+			tb.Helper()
+			rdkStatus := server.Status()
+			test.That(tb, rdkStatus, test.ShouldNotBeNil)
```

Comment:
Maybe a comment above this line saying that we're asserting a non-nil error to ensure the process is dead.